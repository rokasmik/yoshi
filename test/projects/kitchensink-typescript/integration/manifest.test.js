const path = require('path');
const fs = require('fs-extra');
const { ciEnv } = require('../../../../scripts/utils/constants');

jest.setTimeout(60 * 1000);

const originalFilePath = path.join(
  global.scripts.testDirectory,
  'src/client.tsx',
);

const originalContent = fs.readFileSync(originalFilePath, 'utf-8');

async function replaceOriginalEntry(relativeFilePath) {
  const newFilePath = path.join(global.scripts.testDirectory, relativeFilePath);
  const newContent = fs.readFileSync(newFilePath, 'utf-8');

  await fs.writeFile(originalFilePath, newContent);
}

describe('manifest', () => {
  afterEach(() => {
    // reset state back to normal after every test
    fs.writeFileSync(originalFilePath, originalContent);
  });

  it('generates manifest stat file', async () => {
    await replaceOriginalEntry('src/cases/manifest.tsx');

    await global.scripts.build(ciEnv);

    const statsFilePath = path.join(
      process.env.TEST_DIRECTORY,
      'dist/statics/manifest.1.0.0.json',
    );

    const json = JSON.parse(fs.readFileSync(statsFilePath, 'utf-8'));

    expect(json).toMatchSnapshot();
  });
});
