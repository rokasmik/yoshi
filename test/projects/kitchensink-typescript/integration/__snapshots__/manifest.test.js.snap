// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`manifest generates manifest stat file 1`] = `
Object {
  "app": Object {
    "css": Array [
      "https://static.parastorage.com/services/kitchensink/1.0.0/app.min.css",
    ],
    "js": Array [
      "https://static.parastorage.com/services/kitchensink/1.0.0/app.bundle.min.js",
    ],
  },
  "other": Object {
    "js": Array [
      "https://static.parastorage.com/services/kitchensink/1.0.0/other.bundle.min.js",
    ],
  },
}
`;
